dfastmi/batch/core.py

```diff
@@ -483,21 +483,22 @@ def _get_root_dir(rootdir: Path) -> Path:
     return rootdir
 
 
-def count_discharges(discharges: Vector) -> int:
+def count_discharges(apply_q: Tuple[bool, bool, bool]) -> int:
     """
-    Count the number of non-empty discharges.
+    Count the number of discharges to be used (apply_q = True).
 
     Arguments
     ---------
-    discharges : Vector
-        Tuple of (at most) three characteristic discharges (Q).
+    apply_q : Tuple[bool, bool, bool]
+        A list of 3 flags indicating whether each value should be used or not.
+        The Q1 value can't be set to None because it's needed for char_times.
 
     Returns
     -------
     n : int
         Number of non-empty discharges.
     """
-    return sum([q is not None for q in discharges])
+    return sum(apply_q)
 
 
 def get_filenames(
```

Comment:
If this is a legacy method move to legacy.py. Do not do this in this issue (not in scope) but create a new issue to cleanup this file for improved maintenance.

Reply:
Included in issue DFAST-237.